packages/base-subgraph/src/utils/setTokenMetadata.ts

```diff
@@ -0,0 +1,40 @@
+import { Token } from '../../generated/schema'
+import { Bytes, JSONValueKind } from '@graphprotocol/graph-ts'
+import { json } from '@graphprotocol/graph-ts'
+import { decode } from 'as-base64'
+
+export function setTokenMetadata(token: Token, tokenURI: string): void {
+  let split = tokenURI.split(',')
+
+  // Ensure the tokenURI is a data URI
+  if (split.length < 2) return
+
+  // Try to base64 decode the metadata
+  let metadataDecoded = Bytes.fromUint8Array(decode(split[1]))
+
+  // Parse the decoded bytes as JSON
+  let metadataBytes = json.try_fromBytes(metadataDecoded)
+
+  if (metadataBytes.isError) return
+
+  // Cast to object
+  if (metadataBytes.value.kind != JSONValueKind.OBJECT) return
+  let metadataObject = metadataBytes.value.toObject()
+
+  // Set name if present
+  let name = metadataObject.get('name')
+  if (name && name.kind == JSONValueKind.STRING) token.name = name.toString()
+
+  // Set image if present
+  let image = metadataObject.get('image')
+  if (image && image.kind == JSONValueKind.STRING) {
+    let imageString = image.toString()
+    if (imageString.startsWith('https://api.zora.co'))
+      imageString = imageString.replace('https://api.zora.co', 'https://nouns.build/api')
```

Comment:
is there a reason we're replacing the metadata base URI here?

Reply:
This is a straight up copy + paste from nouns builder (commit https://github.com/ourzora/nouns-builder/commit/83e97883d48e099b318a9259ae20f483a2167f60). Since we're using their contracts, we're using a lot of their code for now.